src/script.js

```diff
@@ -86,6 +88,24 @@ $(
       $styleSwitcher.setStyle(styleTitle, $content.contents()[0]);
     });
 
+    var getURLParameter = function(param)
```

Comment:
Do you mean you'd like to (ab)use the hash instead or that you are against the very concept? Because I think passing a search query directly is must-have functionality (and besides it's conceptually a server parameter hijacked by JS ;-P) and I have some application-state related suggestions that I think would seriously improve usability. Much more so than this silly clear or not clear stuff, although I'd love to sneak that setting in there as well. :-)